### DISCUSSION TITLE
I want to capture data in proxy_capture.py and send it to a message queue, and then use app.py to process the queue. proxy_capture.py is started using app.py.

### DISCUSSION THREAD

**Discussion by xirui1998** [mitmproxy/mitmproxy#7310] [Q&A]
### proxy_capture.py
```python
from mitmproxy import http


def log_data(flow: http.HTTPFlow) -> None:
    """Log the request and response related information"""
    if flow.response and flow.response.status_code == 200:
        response_data = {
            "Request URL": flow.request.pretty_url,
            "Response Status Code": flow.response.status_code,
            "Response Text": flow.response.text[:100]
        }
        print(response_data)


def request(flow: http.HTTPFlow) -> None:
    """Handle the request"""
    log_data(flow)


def response(flow: http.HTTPFlow) -> None:
    """Handle the response"""
    log_data(flow)


```

### app.py
```python
import threading
import subprocess
from proxy_switch import ProxySwitch

# Create a global instance of ProxySwitch to avoid multiple initializations
proxy = ProxySwitch()


def open_capture():
    """Start mitmdump and run the specified script"""
    # Open the proxy
    proxy.set_proxy()
    
    # Run the mitmdump script
    script_path = "proxy_capture.py"
    process = subprocess.Popen(["mitmdump", "-s", script_path, "-p", "8083", "-q"])

    try:
        # Keep mitmdump running
        process.wait()
    except KeyboardInterrupt:
        # Handle Ctrl+C interruption to terminate mitmdump
        print("mitmdump interrupted by user.")
        process.terminate()
    except Exception as e:
        # Catch any other exceptions and terminate mitmdump
        print(f"mitmdump terminated with an error: {e}")
        process.terminate()
    finally:
        # Ensure that the proxy is closed when the process ends
        proxy.close_proxy()


if __name__ == '__main__':
    # Start mitmdump capture in a separate thread
    capture_thread = threading.Thread(target=open_capture)
    capture_thread.start()
    capture_thread.join()  # Optionally wait for the thread to finish
```

I tried asking AI, but while data capture can capture the data, the message queue cannot access the data.

**Comment by mhils** [comment]
I'm sorry, but our forums are English only. We don't have enough Chinese contributors to be able to moderate Chinese content.

**Reply by xirui1998** [reply]
Sorry, it has been revised to English content.

**Reply by mhils** [reply]
No problem. I do not understand this statement:

> the message queue cannot access the data.

Can you rephrase?

**Reply by xirui1998** [reply]
proxy_capture.py is my packet capture script file. This script file is started by the open_capture function in app.py. I want to create a queue in the script file proxy_capture.py, put the captured data in the queue, and then process the queue in app.py. The proxy_capture.py file is production and the app.py file is consumption.
